pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/IndexHandler.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.spi.index;
+
+import org.apache.pinot.segment.spi.store.SegmentDirectory;
+
+
+/**
+ * Interface for index handlers, which update the corresponding type of indices,
+ * like adding, removing or converting the format.
+ */
+public interface IndexHandler {
+  /**
+   * Adds new indices and removes obsolete indices.
+   */
+  void updateIndices(SegmentDirectory.Writer segmentWriter)
+      throws Exception;
+
+  /**
+   * Check if there is a need to add new indices or removes obsolete indices.
+   * @return true if there is a need to update.
+   */
+  boolean needUpdateIndices(SegmentDirectory.Reader segmentReader)
```

Comment:
(nit) Put this API above for readability as we always follow the sequence of: check -> update -> post update

Reply:
I guess we are not very strick on that policy given that this file has been movied from package `org.apache.pinot.segment.local.segment.index.loader` to `org.apache.pinot.segment.spi.index` :laughing: 

In this PR it doesn't look like that because I just copied the file in order to do not affect the rest of the code (in case we want to merge this partial PR).

I would suggest to do not change the code in order to make it easier to read the git diff in case we finally move the file from one project to the other.